projects/observability/src/shared/components/explore-query-editor/explore-query-editor.component.ts

```diff
@@ -77,13 +77,13 @@ export class ExploreQueryEditorComponent implements OnChanges, OnInit {
   }
 
   public ngOnChanges(changeObject: TypedSimpleChanges<this>): void {
-    if (changeObject.filters) {
-      this.visualizationBuilder.filters(this.filters);
-    }
-
     if (changeObject.context) {
       this.visualizationBuilder.context(this.context);
     }
+
+    if (changeObject.filters) {
+      this.visualizationBuilder.filters(this.filters);
```

Comment:
Since visualizationBuilder is a builder, how does the preference matter?

Reply:
It's a bit confusing api, but it basically exposes an observable which emits on each change in the builder - changing the context resets the whole thing though, so we'd want that to happen first.

Reply:
👆